### PR TITLE
Modifying `<head>`: clarity and placement in the docs

### PR DESCRIPTION
**Pull request by Jutanium**
#### TL;DR

- Moves the _Modifying `<head>`_ section into the Layouts section of `/core-concepts/astro-pages`
- Clarifies that there is no special handling of the `<head>` tag

#### What kind of changes does this PR include?

- New or updated content

#### Description

This addresses #1378.

The behavior of the `<head>` tag has little to do with the difference between JSX and Astro components. It does have to do with certain libraries within the React ecosystem, but that is not due to the JSX syntax. Documenting `<head>` makes more sense in the Layouts section of `Astro Pages`, because we recommend using the tag in a Layout, so I moved it there. (I could also see moving this to its own section within `Astro Pages`).

The section itself was unclear, as #1378 describes. Astro does not have any special handling of `<head>`, and I updated the wording to make that more clear.

### CODE REVIEW

**Comment by sarah11918** [comment]
LGTM @Jutanium! Great call!
